### ISSUE TITLE
@cputarget attribute

### ISSUE THREAD

**Issue by lerno** [c3lang/c3c#599]
Set the attribute, e.g. `@cputarget("sse3")` to compile a particular function for a particular featureset.

**Comment by lerno** [comment]
Possibilities:
```c
fn void foo() @target("sse4.1") { ... }
fn void foo() @target(sse4.1) { ... }
```
If string:
```c
fn void foo() @target("v1+sse4.1+ssse3-mmx") { ... }
fn void foo() @target("cpu=v1,+sse4.1,+ssse3,-mmx") { ... }
fn void foo() @target(.cpu = "v1",.features = "sse4.1,+ssse3,-mmx") { ... }
fn void foo() @target("v1", "sse4.1", "ssse3", !"mmx") { ... }
```
If not:
```c
fn void foo() @target(v1, sse4.1, ssse3, !mmx) { ... }
fn void foo() @target(v1: +sse4.1, +ssse3, -mmx) { ... }
fn void foo() @target(v1) @features(sse4.1, ssse3, !mmx) { ... }
```

**Comment by data-man** [comment]
I think it will useful for asm-functions only.

**Comment by lerno** [comment]
It's intended for being able to write for multiple targets in the same compilation. ASM will always ignore instruction limitations.

**Comment by lerno** [comment]
For the latter, like you can use CPUID to switch between calling function A or B depending on what CPU is detected at startup.

**Comment by data-man** [comment]
Yes. As a former Delphi/FreePascal developer I know that their runtime can use CPU capability detection to override memory manager [fastmove](https://github.com/fpc/FPCSource/blob/main/rtl/i386/fastmove.inc) functions. :)

**Comment by lerno** [comment]
Yeah, so this was the idea. I had some thoughts of doing "target_clone" from GCC, but I think I limit myself to just "target" as there are some challenges making it work seamlessly.
